python/shotgun_fields/image_widget.py

```diff
@@ -279,24 +318,53 @@ def _replace_image(self):
         """
         self._upload_image()
 
-    def _scale_pixmap(self):
+    def _scale_pixmap(self, pixmap):
         """
         Scale the pixmap in preparation for display.
         """
-        scaled_pixmap = self._pixmap.scaled(
+        return pixmap.scaled(
             self.size(),
             QtCore.Qt.KeepAspectRatio,
             QtCore.Qt.SmoothTransformation
         )
-        self._scaled_width = scaled_pixmap.width()
-        super(ImageWidget, self).setPixmap(scaled_pixmap)
-        self._update_btn_position()
 
     def _show_image(self):
         """
-        The user requested to show the file. Forward to desktop services.
+        The user requested to show the file.
+
+        Display the image in a transient, modeless QDialog.
         """
-        QtGui.QDesktopServices.openUrl("file:///%s" % (self._image_url))
+
+        # don't continue unless there's somethign to show
+        if not self._pixmap:
+            return
+
+        if self._image_url:
```

Comment:
reading the code, looks like `self._image_url` is actually a path to a local file on disk? Url makes me think it's a `http://...` type url. I think this is confusing - why not say image_path or something? sorry if i misread the code?

One of the reasons to have extra clarity here is because thumb urls are S3 urls with expiry, so they are not valid more than a couple of minutes and should never be serialized or stored, even in memory.